planner/core/plan_cost.go

```diff
@@ -172,7 +174,37 @@ func (p *PhysicalTableReader) GetPlanCost(taskType property.TaskType) (float64,
 
 // GetPlanCost calculates the cost of the plan if it has not been calculated yet and returns the cost.
 func (p *PhysicalIndexMergeReader) GetPlanCost(taskType property.TaskType) (float64, error) {
-	return 0, errors.New("not implemented")
+	if p.planCostInit {
+		return p.planCost, nil
+	}
+	p.planCost = 0
+	netFactor := p.ctx.GetSessionVars().GetNetworkFactor(nil)
+	if tblScan := p.tablePlan; tblScan != nil {
+		childCost, err := tblScan.GetPlanCost(property.CopSingleReadTaskType)
+		if err != nil {
+			return 0, err
+		}
+		p.planCost += childCost // child's cost
+		tblStats := getTblStats(tblScan)
+		rowSize := tblStats.GetAvgRowSize(p.ctx, tblScan.Schema().Columns, false, false)
+		p.planCost += tblScan.StatsCount() * rowSize * netFactor // net I/O cost
+	}
+	for _, idxScan := range p.partialPlans {
+		childCost, err := idxScan.GetPlanCost(property.CopSingleReadTaskType)
+		if err != nil {
+			return 0, err
+		}
+		p.planCost += childCost // child's cost
+		tblStats := getTblStats(idxScan)
+		rowSize := tblStats.GetAvgRowSize(p.ctx, idxScan.Schema().Columns, true, false)
```

Comment:
One tiny difference with the old implementation is that the old implementation differentiates partial table scan with partial index scan. When calculating the cost of partial table scan, the argument `isEncodedKey` in `GetAvgRowSize` is set to false.

Reply:
You are right. Fixed, PTAL

Reply:
Is this fixed?

Reply:
And I think we should add some test cases for this case.

Reply:
Added, PTAL https://github.com/pingcap/tidb/pull/34169/commits/b60f01e326005b93179a9fcc402383889a40ae47